satellite/satellitedb/overlaycache.go

```diff
@@ -22,6 +24,243 @@ type overlaycache struct {
 	db *dbx.DB
 }
 
+type filterReputableRequest struct {
```

Comment:
I feel like you could use the same struct for both of these, and just don't use `minReputation` for your `filterNewNodes` function. That way you don't need two separate structs and you can keep everything else the same pretty uch